docs/development/index.rst

```diff
@@ -0,0 +1,41 @@
+:orphan:
+
+***********************
+Contributing to Astropy
```

Comment:
Since you invited bikeshedding... :smile: This feels like it should just be `Developer Documentation`, following the section title in the main index page. `Contributing to Astropy` is a much broader thing (https://www.astropy.org/contribute.html), while the docs here are really focused on developing code for the astropy core package.

Reply:
That's why this is still a WIP.  My next step that I've been working on is making the page (at least at the top level) a little more generic.  My inspiration here is matplotlib (which incidentally does still name itself "The Developer's Guide"): https://matplotlib.org/stable/devel/index.html

But I'm happy to revisit this once the PR is further along.